internal/discover/hooks.go

```diff
@@ -100,7 +109,7 @@ type HookCreator interface {
 	Create(HookName, ...string) *Hook
 }
 
-// WithDisabledHooks sets the set of hooks that are disabled for the CDI hook creator.
+// WithDisabledHooks explicitly disables the specified hooks.
 // This can be specified multiple times.
 func WithDisabledHooks(hooks ...HookName) Option {
```

Comment:
Let's update these to also use generics so that we can pass `...string` or `...HookName`.

Reply:
I was too hasty in suggesting this. As discussed offline, I think a strongly-typed internal API is fine since we only every call this with `HookName` arguments. If we need more flexibility in the future we can add generics.